File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.java
// Last updated: 2/3/2026, 9:40:02 PM
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int pre =1;
        int pos=1;
        int[] res=new int[nums.length];
        Arrays.fill(res,1);
        for(int i=0;i<nums.length;i++){
            res[i]*=pre;
            pre*=nums[i];
        }
        for(int i=nums.length-1;i>=0;i--){
            res[i]*=pos;
            pos*=nums[i];
        }
        return res;
    }
}

File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.py
# Last updated: 2/3/2026, 9:39:48 PM
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        length=len(nums)
        l_arr=[0]*length
        r_arr=[0]*length
        l_value=1
        r_value=1
        
        for i in range(length):
            rev=-i-1
            l_arr[i]=l_value
            r_arr[rev]=r_value
            l_value*=nums[i]
            r_value*=nums[rev]

        return [i*j for i,j in zip(l_arr,r_arr)]

        